libfreedo/arm.cpp

```diff
@@ -600,14 +600,37 @@ unsigned char * _arm_Init()
 
 void _arm_Destroy()
 {
-	io_interface(EXT_WRITE_NVRAM,pNVRam);//_3do_SaveNVRAM(pNVRam);
+    io_interface(EXT_WRITE_NVRAM,pNVRam);//_3do_SaveNVRAM(pNVRam);
 
+    if (profiling != NULL) {

```

Comment:
there must be something causing this method to be called on different threads or multiple times - this should generally be safer now